src/DotNetBoilerplate.Core/Employees/Employee.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+
+namespace DotNetBoilerplate.Core.Employees
+{
+    public class Employee
+    {
+        private Employee(Guid id, Guid organizationId)
+        {
+            Id = id;
+            UserId = id;
+            OrganizationId = organizationId;
+            Role = RoleInOrganization.None;
+        }
+
+        public Employee()
+        {
+            Id = Guid.NewGuid();
+            UserId = Guid.Empty; 
+            OrganizationId = Guid.Empty;
+            Role = RoleInOrganization.None;
+        }
+
+        public Guid Id { get; private set; }
+        public Guid UserId { get; private set; }
+        public Guid OrganizationId { get; private set; }
+        public RoleInOrganization Role { get; private set; }
+
+        public enum RoleInOrganization
+        {
+            None, // Defaultowy użytkownik bez uprawnień
+            Moderator, // Uprawnienia do może dodawawania pracowników do projektu, może dodawać i edytować projekty
+            Admin // Zarządza organizacją, projektami i ich członkami
+        }
+
+        public void UpdateRole(RoleInOrganization roleInOrganization)
+        {
+            if (roleInOrganization == RoleInOrganization.None) { Role = RoleInOrganization.None; }
+            else if (roleInOrganization == RoleInOrganization.Moderator) { Role = RoleInOrganization.Moderator; }
+            else if (roleInOrganization == RoleInOrganization.Admin) { Role = RoleInOrganization.Admin; }
+        }
+
+        public static Employee Create(Guid userId, Guid organizationId, RoleInOrganization role)
+        {
+            if (organizationId != Guid.Empty)
+                throw new Exception("User already belongs to organization");
```

Comment:
Customowy sobie zróbmy exceotuib

Reply:
Poza tym tutaj ten warynek trochę bez sensu bo jak mamy statyczną metodę to pewnie będziemy używać na jak statyczną czyli bez instancji, no a bez instancji organization id będznie zawsze nullem bo nigdzie go nie przypiszemy